core/src/main/java/com/valtech/aem/saas/core/autocomplete/DefaultAutocompleteService.java

```diff
@@ -56,10 +57,13 @@
                 getApiUrl(index) + getQueryString(text, language,
                                                   getEffectiveFilters(searchConfiguration.getFilters(), filters)));
         return searchApiRequestExecutorService.execute(searchRequestGet)
-                .filter(SearchResponse::isSuccess)
-                .flatMap(response -> response.get(new TypeaheadDataExtractionStrategy(
-                        language)))
-                .orElse(Collections.emptyList());
+                                              .filter(SearchResponse::isSuccess)
+                                              .flatMap(response -> response.get(new TypeaheadDataExtractionStrategy(
+                                                      language)))
+                                              .orElse(Collections.emptyList())
+                                              .stream()
+                                              .limit(searchConfiguration.getAutocompleteResultsMaxTotal())
+                                              .collect(Collectors.toList());
```

Comment:
**Thought**

That looks like a sql query :D 

In javascript, you can get x elements with filter using the index, not sure in Java if that's the same filter function though.

myArray.filter((element, index) => index < MAX), but yeah you still go through the entire array.
I was just wondering. Nothing wrong with using limit I guess.

Reply:
It is the java Stream api (https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html).
Limit is applied to a stream created from the collection of results, and it iterates until the specified limit position.